pennylane/ftqc/lattice.py

```diff
@@ -0,0 +1,157 @@
+    ):
+        self._lattice_shape = lattice_shape
+        if graph is None:
+            if nodes is None and edges is None:
+                raise ValueError(
+                    "Neither a networkx Graph object nor nodes together with edges are provided."
+                )
+            self._graph = nx.Graph()
+            self._graph.add_nodes_from(nodes)
+            self._graph.add_edges_from(edges)
+        else:
+            self._graph = graph
+
+    @property
+    def get_lattice_shape(self) -> str:
+        r"""Returns the lattice shape name."""
+        return self._lattice_shape
+
+    def get_neighbors(self, node):
+        r"""Returns the neighbors of a given node in the lattice.
+
+        Args:
+            node: a target node label.
+        """
+        return self._graph.neighbors(node)
+
+    @property
+    def get_nodes(self):
+        r"""Returns all nodes in the lattice."""
+        return self._graph.nodes
+
+    @property
+    def get_edges(self):
+        r"""Returns all edges in the lattice."""
+        return self._graph.edges
+
+    @property
+    def get_graph(self) -> nx.Graph:
+        r"""Returns the underlying NetworkX graph object representing the lattice."""
+        return self._graph
+
+
+class LatticeShape(Enum):
+    """Enum to define valid set of lattice shape supported."""
+
+    chain = 1
+    square = 2
+    rectangle = 3
+    triangle = 4
+    honeycomb = 5
+    cubic = 6
+
+
+# map between lattice name and dimensions
+_LATTICE_DIM_MAP = {
+    "chain": 1,
+    "square": 2,
+    "rectangle": 2,
+    "cubic": 3,
+    "triangle": 2,
+    "honeycomb": 2,
+}
+
+# map between lattice name and networkx method name
+_LATTICE_GENERATOR_MAP = {
+    "chain": "grid_graph",
+    "square": "grid_graph",
+    "rectangle": "grid_graph",
+    "cubic": "grid_graph",
+    "triangle": "triangular_lattice_graph",
+    "honeycomb": "hexagonal_lattice_graph",
+}
+
+
+def generate_lattice(dims: list[int], lattice: str) -> Lattice:
+    r"""Generates a :class:`~pennylane.ftqc.Lattice` object with a given lattice dimensions and its shape name.
+
+    Args:
+        dims(List[int]): Dimensions for lattice generation. For lattices generated by `nx.grid_graph` ( ``'chain'``, ``'rectangle'``,  ``'square'``, ``'cubic'``),
+        `dims` contains the number of nodes in the each direction of grid. Per ``'honeycomb'`` or ``'triangle'``, the generated lattices will have dims[0] rows and dims[1]
+        columns of hexagons or triangles.
+        lattice (str): Shape of the lattice. Input values can be ``'chain'``, ``'square'``, ``'rectangle'``, ``'honeycomb'``, ``'triangle'``, ``'cubic'``.
+
+    Returns:
+        a :class:`~pennylane.ftqc.Lattice` object.
+
+    Raises:
+        ValueError: If the lattice shape is not supported or the dimensions are invalid.
+    """
+
+    lattice_shape = lattice.strip().lower()
+    supported_shape = [shape.name for shape in LatticeShape]
```

Comment:
Should we define this outside the function call, so we can populate it once and reuse it ?